src/summary/documentation/vital-card-level-two.component.tsx

```diff
@@ -0,0 +1,175 @@
+import React from "react";
+import { match } from "react-router";
+import { performPatientsVitalsSearch } from "./vitals-card.resource";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
+import { formatDate } from "./dimension-helpers";
+import styles from "./vital-card-level-two.css";
+import SummaryCard from "../cards/summary-card.component";
+import { useCurrentPatient } from "@openmrs/esm-api";
+
+export default function VitalsLevelTwo(props: VitalsLevelTwoProps) {
+  const resultsPerPage = 15;
+
+  const [patientVitals, setPatientVitals] = React.useState(null);
+  const [totalPages, setTotalPages] = React.useState(1);
+  const [currentPage, setCurrentPage] = React.useState(1);
+  const [showNextButton, setShowNextButton] = React.useState(false);
+  const [showPreviousButton, setShowPreviousButton] = React.useState(false);
+  const [pagedResults, setPagedResults] = React.useState([]);
+  const [
+    isLoadingPatient,
+    patient,
+    patientUuid,
+    patientErr
+  ] = useCurrentPatient();
+
+  React.useEffect(() => {
+    if (!isLoadingPatient && patient) {
+      const subscription = performPatientsVitalsSearch(patient.id).subscribe(
+        vitals => {
+          setPatientVitals(vitals);
+          setTotalPages(Math.ceil(vitals.length / resultsPerPage));
+          setPagedResults(vitals.slice(0, resultsPerPage));
+        },
+        createErrorHandler()
+      );
+
+      return () => subscription.unsubscribe();
+    }
+  }, [isLoadingPatient, patient]);
+
+  React.useEffect(() => {
+    {
+      patientVitals && currentPage * resultsPerPage > patientVitals.length
+        ? setShowNextButton(false)
+        : setShowNextButton(true);
+      currentPage !== 1
+        ? setShowPreviousButton(true)
+        : setShowPreviousButton(false);
+    }
+  }, [pagedResults, currentPage, patientVitals]);
+
+  const nextPage = () => {
+    let upperBound = currentPage * resultsPerPage + resultsPerPage;
+    const lowerBound = currentPage * resultsPerPage;
+    if (upperBound > patientVitals.length) {
+      upperBound = patientVitals.length;
+    }
+    const pageResults = patientVitals.slice(lowerBound, upperBound);
+    setPagedResults(pageResults);
+    setCurrentPage(currentPage + 1);
+  };
+
+  const previousPage = () => {
+    const lowerBound = currentPage * resultsPerPage - resultsPerPage * 2;
+    const upperBound = currentPage * resultsPerPage - resultsPerPage;
```

Comment:
I prefer `resultsPerPage * (currentPage - 2)` and `resultsPerPage * (currentPage - 1)`, but that's totally personal preference and they are of course equivalent.

Reply:
Got this fixed